atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -591,6 +593,67 @@ public void testSweepTimestampMetric() {
         TargetedSweepMetricsTest.assertSweepTimestampThoroughEquals(5);
     }
 
+    @Test
+    public void doNotSweepAnythingAfterEntryWithCommitTsAfterSweepTs() {
+        immutableTs = 1000L;
+        enqueueWrite(TABLE_CONS, 900);
+        enqueueWrite(TABLE_CONS, 910);
+        enqueueWriteCommitedAt(TABLE_CONS, 950, 1000);
+        enqueueWrite(TABLE_CONS, 970);
+        enqueueWrite(TABLE_CONS, 920);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(920L);
+        ArgumentCaptor<Map> argument = ArgumentCaptor.forClass(Map.class);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(eq(TABLE_CONS), argument.capture(), eq(false));
+        assertThat(argument.getValue()).containsValue(920L);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(920L);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap(), eq(false));
+
+        immutableTs = 1101L;
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(1101L - 1L);
+        verify(spiedKvs, times(2)).deleteAllTimestamps(eq(TABLE_CONS), argument.capture(), eq(false));
+        assertThat(argument.getValue()).containsValue(970L);
+    }
+
+    @Test
+    public void doNotDeleteAnythingAfterEntryWithCommitTsAfterSweepTs() {
+        immutableTs = 1000L;
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONS, 900), 900);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONS, 920), 920);
+        enqueueWriteCommitedAt(TABLE_CONS, 950, 2000);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONS, 970), 970);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONS, 1110), 1110);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(920L);
+        verify(spiedKvs, never()).deleteAllTimestamps(any(TableReference.class), anyMap(), eq(false));
+
+        ArgumentCaptor<Multimap> multimap = ArgumentCaptor.forClass(Multimap.class);
+        verify(spiedKvs, times(1)).delete(eq(TABLE_CONS), multimap.capture());
```

Comment:
Why were 900 and 920 deleted directly rather than via `deleteAllTimestamps`? From the code on this PR, the direct delete should only occur if the transactions weren't committed, and it looks like they should be.

nit: The two new tests use  `enqueueWrite`, `sweepQueue.enqueue`, and `enqueueWriteCommittedAt`, which are hard to reason about given that I assume they must behave differently, but seem very similar.

Reply:
Definitely wasn't clear enough. Should be good now

Reply:
Looks good now, thanks.